app/controllers/AdminloginController.php

```diff
@@ -3,6 +3,31 @@
 
 class AdminloginController extends Base
 {
+    private static function getHashedPassword($password)
+    {
+        return sha1($password);
+    }
+
+    private static function isCorrectPassword($exceptedPassword, $givenPassword)
+    {
+        return $exceptedPassword == self::getHashedPassword($givenPassword);
+    }
+
+    private static function isCorrectCredentials($config, $credentialsGiven)
+    {
+        return (
+           isset($credentialsGiven['username'])
+        && isset($credentialsGiven['password'])
+        && $config->username == $credentialsGiven['username']
+        && self::isCorrectPassword($config->password, $credentialsGiven['password'])
+        );
+    }
+
+    private static function isCorrectCredentialsFromPost($config)
+    {
+        return self::isCorrectCredentials($config, $_POST);
```

Comment:
There's a getPost method in the parent, maybe we should use that?

Reply:
`cd app/ && grep -r 'class Base'` returns nothing. Is it in Modl, Moxl, or somewhere else?